scripts/staging/fm-binclass.dml

```diff
@@ -0,0 +1,181 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+
+/*
+ * Factorization Machines for binary classification.
+ */
+
+# Imports
+source("nn/optim/adam.dml") as adam
+source("nn/layers/fm.dml") as fm
+source("nn/layers/log_loss.dml") as log_loss
+source("nn/layers/sigmoid.dml") as sigmoid
+source("nn/layers/l2_reg.dml") as l2_reg
+source("nn/layers/cross_entropy_loss.dml") as cross_entropy_loss
+
+train = function(matrix[double] X, matrix[double] y, matrix[double] X_val, matrix[double] y_val)
+    return (matrix[double] w0, matrix[double] W, matrix[double] V, matrix[double] loss) {
+  /*
+   * Trains the FM model.
+   *
+   * Inputs:
+   *  - X : n examples with d features, of shape (n, d).
+   *  - y : label corresponds to n examples  
+   *  - lambda : regularization (5e-04)
+   * 
+   * Outputs:
+   *  - w0, W, V : updated model parameters.
+   *  - loss : computed loss with log_loss.
+   */
+
+  /* input propagation through layers
+   * fm::init -> adam::init -> fm::forward -> sigmoid::forward -> log_loss::forward \
+   *          adam::update <- fm::backward <- sigmoid::backward <- log_loss::backward <-
+   */
+
+ 
+    # 1.initialize fm core
+    [w0, W, V] = fm::init(d, k);
```

Comment:
this won't run unless you determine `d` within this function, and have `k` passed in as an argument to `train`.  I would highly advise trying to run these functions, as it will make any issues like this readily apparent.